### DISCUSSION TITLE
Rails jRuby Postgres trying to connect to Redshift?

### DISCUSSION THREAD

**Discussion by Tim-Blokdijk** [jruby/jruby#8042] [Q&A] [answered]
When updating my jRuby Rails project (probably the activerecord-jdbcpostgresql-adapter gem) I get the following error when trying to start the server:
`FATAL -- : NoMethodError (undefined method "database_product" for nil:NilClass) redshift = !! (@connection.database_product || '').index('Redshift')`

Is this a known error?
I also got this when creating a new jRuby Postgresql based Rails project.

**Comment by enebo** [comment, marked as answer]
@Tim-Blokdijk Which version of Rails?  We do not support running Rails 7.1 at all yet.  Earlier versions should work but I don't know what redshift is so there could be something that version of postgres does that we don't handle yet.

**Reply by Tim-Blokdijk** [reply]
I did a bundle update, which updated to Rails 7.1, but if it's not supported yet then it's the expected behavior. 
Apparently Redshift is a PostgreSQL version maintained by Amazon. So for some reason it drops in a codepath that tries to connect to that instead of my locally running PostgreSQL.
Maybe it would be good to add an explicit warning that jRuby 9.4.5.0 is known not to work with Rails 7.1 and it's only supported with Rails 7.0 when run with Rails 7.1 or up?

Anyway, thanks for the answer.
